### DISCUSSION TITLE
Cypress and clickjacking (iframe)

### DISCUSSION THREAD

**Discussion by portsoller** [cypress-io/cypress#9615] [Questions and Help]
Hi to all.
Now i  have a problem with my cypress project.
Lately our developers has added clickjacking to some pages of our project and now my tests cannot work.
If some our pages opened in iframe, page say that user must "Follow to login page." and etc.
https://prnt.sc/vztppq
how can i fix my tests for work with some our pages of project (signup, login, dashboard and etc)
Thanks for help

**Comment by portsoller** [comment]
Already found some way:
    Cypress.on('window:before:load', (win) => {
      Object.defineProperty(win, 'self', {
        get: () => {
          return window.top
        }
      })
    })


but, after filling the fields and during send the form of registration, page state return to default. 
Think need to change this code.

**Comment by jennifer-shehane** [comment]
Cypress runs your application in an iframe, which is exactly what clickjacking code tries to prevent. So your application is preventing itself from running within Cypress. We try to look for clickjacking code and remove it so that app run fine in Cypress: https://on.cypress.io/configuration#modifyObstructiveCode

The best way around this would be to remove the clickjack code when you know you're running within Cypress: https://on.cypress.io/using-cypress-faq#Is-there-any-way-to-detect-if-my-app-is-running-under-Cypress

**Reply by portsoller** [reply]
Our project manager strongly requested that this security (clickjacking) be added. Now that day I have been engaged not only in writing new tests, but in studying how to fix the existing ones.

**Reply by portsoller** [reply]
The above code helped me open the registration or login pages. But now a new problem has arisen. after entering valid authorization data, when sending, the server replies that the fields (for example) email should not be empty. while the data in the field is entered
http://prntscr.com/w0edrp
